src/ServiceControl.Plugin.Nsb6.Heartbeat/Heartbeats.cs

```diff
@@ -82,33 +85,33 @@ protected override void OnStop()
                     cancellationTokenSource.Cancel();
                 }
 
-                base.OnStop();
+                return Task.FromResult(0);
             }
 
-            void NotifyEndpointStartup(HostInformation hostInfo, DateTime startupTime)
+            void NotifyEndpointStartup(DateTime startupTime)
             {
                 // don't block here since StartupTasks are executed synchronously.
-                Task.Run(() => SendEndpointStartupMessage(hostInfo, startupTime, cancellationTokenSource.Token));
+                Task.Run(() => SendEndpointStartupMessage(startupTime, cancellationTokenSource.Token));
```

Comment:
We don't need this `Task.Run` just don't await the Task returned from `SendEndpointStartupMessage` and use `TaskEx.Ignore()`